docs/source/access.mdx

```diff
@@ -78,4 +84,34 @@ Slicing returns a slice - or subset - of the dataset, which is useful for viewin
  'text': ['if you sometimes like to go to the movies to have fun , wasabi is a good place to start .',
   "emerges as something rare , an issue movie that's so honest and keenly observed that it doesn't feel like one .",
   'the film provides some great insight into the neurotic mindset of all comics -- even those who have reached the absolute top of the game .']}
-```
\ No newline at end of file
+```
+
+## IterableDataset
+
+An [`IterableDataset`] is loaded when you set the `streaming` parameter to `True` in [`~datasets.load_dataset`]:
+
+```py
+>>> from datasets import load_dataset
+
+>>> iterable_dataset = load_dataset("food101", split="train", streaming=True)
+>>> for example in iterable_dataset:
+...     print(example)
+...     break
+{'image': <PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=384x512 at 0x7F0681F5C520>, 'label': 6}
+```
+
+An [`IterableDataset`] progressively iterates over a dataset one example at a time, so you don't have to wait for the whole dataset to download before you can use it. As you can imagine, this is quite useful for large datasets you want to use immediately!
+
+However, this means an [`IterableDataset`]'s behavior is different from a regular [`Dataset`]. You don't get random access to examples in an [`IterableDataset`]. Instead, you can call `next(iter()` to return the next item from the [`IterableDataset`]:
```

Comment:
Nit: strictly speaking, elements can be accessed using the iterable protocol; not necessarily only `next(iter())`, also can be iterated using for loops (like in the example above).
```suggestion
However, this means an [`IterableDataset`]'s behavior is different from a regular [`Dataset`]. You don't get random access to examples in an [`IterableDataset`]. Instead, you should iterate over its elements, for example by calling `next(iter())` to return the next item from the [`IterableDataset`]:
```

Reply:
I'd also suggest to use a for loop instead if mentioning `next()` here.

You can also show how to get the first 5 examples though:
```python
print(list(iterable_dataset.take(5)))
```